### PR TITLE
[Dubbo-issue-2184] Fix Return the hanging client requests ASAP. when connection is broken unexpectedly. #2184

### PR DESCRIPTION
**Pull request by daryl666**
## What is the purpose of the change

Fix Return the hanging client requests ASAP. when connection is broken unexpectedly. #2184

## Brief changelog

add the configuration of heartbeat reconnect, default true, if set false, when heartbeat timeout, consumer disconnect

## Verifying this change

org.apache.dubbo.common.Constants
org.apache.dubbo.remoting.exchange.support.header.HeartBeatTask

From Baiji, Term 268, Team 3, Issue #2184, Q3-5

### CODE REVIEW

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/HeartBeatTask.java [hunk 1]
```diff
@@ -70,7 +71,8 @@ public void run() {
                     if (lastRead != null && now - lastRead > heartbeatTimeout) {
                         logger.warn("Close channel " + channel
                                 + ", because heartbeat read idle time out: " + heartbeatTimeout + "ms");
-                        if (channel instanceof Client) {
+
+                        if (channel instanceof Client && channel.getUrl().getParameter(Constants.HEARTBEAT_RECONNECT, Constants.DEFAULT_HEARTBEAT_RECONNECT)) {
```
Comment (carryxyh) [inline dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/HeartBeatTask.java]: This op can not make the hanging request return directly. It just control the reconnect.
I didn't think this pr can solve this problem.
Maybe u should fix or send another pr 
:)

Comment (chickenlj) [reply dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/HeartBeatTask.java]: Agree. This patch didn't solve the problem
